### ISSUE TITLE
Add spoiler tag

### ISSUE THREAD

**Issue by bdharrington7** [akhoury/bbcode-to-markdown#2]
Hey Aziz, almost same kind of dillemma as with the html-to-markdown

Just wanted to add the [spoiler] tag, but if you can point me in the right direction I can make a PR

Thanks!

**Comment by akhoury** [comment]
can you give me an example of the spoiler tag? with all the possible attributes and content?

**Comment by bdharrington7** [comment]
it's simply `[spoiler]` and `[/spoiler]`, no other attributes, but the content inside can be pretty much anything

**Comment by akhoury** [comment]
i think it's already supported, as an ContentOnlyTag

see this line,
https://github.com/akhoury/bbcode-to-markdown/blob/master/newTags.js#L28

**Comment by bdharrington7** [comment]
Ok. Was that intended to just strip it out? I thought it would replace with markdown, like `[b]bold[/b]` would become `**bold**`. in a few of the tags I tested (including spoiler), the output was:

```
bbcode('[u]underline[/u]') //'underline'
bbcode('[vimeo]underline[/vimeo]') //'underline'
```

Could we make the spoiler tag output this to work with @psychobunny's spoiler plugin?

```
bbcode('[spoiler] something secret [/spoiler]') //  '>! something secret'
```

**Comment by bdharrington7** [comment]
Ah I see, you're converting to HTML first, and there's not really an equivalent translation. Wouldn't it be safe to convert straight to the quote md syntax?

**Comment by akhoury** [comment]
yea, i guess that's possible. I mean, `>!` is still valid html

**Comment by akhoury** [comment]
OR, we create custom html tag, like a `<span original-bbcode-tag="spoiler">CONTENT</span>` this way if someone want to just convert to HTML, it won't affect the HTML content, but the markdown converter can recognize it

**Comment by akhoury** [comment]
so i decided on converting

``` bbcode
[spoiler foo=bar]BAMM[/spolier]
```

To

``` html
<spoiler foo="bar">BAMM</spolier>
```

Which, as HTML, is not really valid, but the browser will be still display it as a text node.

Then in this plugin, we need to add support for `<spoiler>`
https://github.com/akhoury/html-md-optional_window/issues/3

I just gotta brush up on coffeescript

**Comment by akhoury** [comment]
basically adding anything in the simpleTags Array will be handled that way
https://github.com/akhoury/bbcode-to-markdown/blob/master/newTags.js#L8

**Comment by akhoury** [comment]
published `html-md-optional_window@3.0.4`
